src/providers/postgres/qgspostgresprovider.cpp

```diff
@@ -591,6 +591,29 @@ QString QgsPostgresUtils::whereClause( QgsFeatureId featureId, const QgsFields &
 
 QString QgsPostgresUtils::whereClause( const QgsFeatureIds &featureIds, const QgsFields &fields, QgsPostgresConn *conn, QgsPostgresPrimaryKeyType pkType, const QList<int> &pkAttrs, const std::shared_ptr<QgsPostgresSharedData> &sharedData )
 {
+  auto lookupKeyWhereClause = [ = ]( bool pkIsString )
+  {
+    if ( featureIds.isEmpty() )
+      return QString();
+
+    //simple primary key, so prefer to use an "IN (...)" query. These are much faster then multiple chained ...OR... clauses
+    QString delim;
+    QString expr = QStringLiteral( "%1 IN (" ).arg( QgsPostgresConn::quotedIdentifier( fields.at( pkAttrs[0] ).name() ) );
+
+    for ( const QgsFeatureId featureId : std::as_const( featureIds ) )
+    {
+      QVariantList pkVals = sharedData->lookupKey( featureId );
+      if ( !pkVals.isEmpty() )
+      {
+        expr += delim + ( pkIsString ? QgsPostgresConn::quotedValue( pkVals[0].toString() ) : pkVals[0].toString() );
```

Comment:
```suggestion
        expr += delim + ( pkIsString ? QgsPostgresConn::quotedValue( pkVals.at( 0 ).toString() ) : pkVals.at( 0 ).toString() );
```

Especially in combination with a non-const list this risks a detach (because the `[]` operator can modify the list and add new items).

Reply:
thanks for the hint, I'll fix it!